pygmt/src/nearneighbor.py

```diff
@@ -0,0 +1,154 @@
+"""
+nearneighbor - Grid table data using a "Nearest neighbor" algorithm
+"""
+
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    data_kind,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+from pygmt.io import load_dataarray
+
+
+@fmt_docstring
+@use_alias(
+    E="empty",
+    G="outgrid",
+    I="spacing",
+    N="sectors",
+    R="region",
+    S="search_radius",
+    V="verbose",
+    a="aspatial",
+    b="binary",
+    d="nodata",
+    e="find",
+    f="coltypes",
+    g="gap",
+    h="header",
+    i="incols",
+    r="registration",
+    w="wrap",
+)
+@kwargs_to_strings(R="sequence", i="sequence_comma")
+def nearneighbor(data=None, x=None, y=None, z=None, **kwargs):
+    r"""
+    Grid table data using a "Nearest neighbor" algorithm
+
+    **nearneighbor** reads arbitrarily located (*x,y,z*\ [,\ *w*]) triples
+    [quadruplets] and uses a nearest neighbor algorithm to assign a weighted
+    average value to each node that has one or more data points within a search
+    radius centered on the node with adequate coverage across a subset of the
+    chosen sectors. The node value is computed as a weighted mean of the
+    nearest point from each sector inside the search radius. The weighting
+    function and the averaging used is given by:
+
+    .. math::
+        w(r_i) = \frac{{w_i}}{{1 + d(r_i) ^ 2}},
+        \quad d(r) = \frac {{3r}}{{R}},
+        \quad \bar{{z}} = \frac{{\sum_i^n w(r_i) z_i}}{{\sum_i^n w(r_i)}}
+
+    where :math:`n` is the number of data points that satisfy the selection
+    criteria and :math:`r_i` is the distance from the node to the *i*'th data
+    point. If no data weights are supplied then :math:`w_i = 1`.
+
+    .. figure:: :gmt-docs:`_images/GMT_nearneighbor.png`
```

Comment:
I changed this line in c22ebb0f2502b2d8189e7eb27416ec20b56308b9 because it was longer than 79 characters, but now the image isn't showing up. Any idea how to make the image show up without using the long URL https://docs.generic-mapping-tools.org/latest/_images/GMT_nearneighbor.png?

```suggestion
    .. figure:: https://docs.generic-mapping-tools.org/latest/_images/GMT_nearneighbor.png
```

Reply:
I do not know the answer to your question, but one option would be to write a PyGMT tutorial for this after the feature is merged rather than including the image in the docstring.

Reply:
> I do not know the answer to your question, but one option would be to write a PyGMT tutorial for this after the feature is merged rather than including the image in the docstring.

I'm a big fan of this option!

Reply:
Yeah, but even in the tutorial .py file, the URL length would still exceed 79 characters and result in a lint error :sweat_smile:

Reply:
To clarify, I was imagining that we could create our own explanatory figure in the PyGMT tutorial rather than including the GMT version.

I do not have a problem with keeping the image link for now since it would be great to merge in this feature PR, but I think long term we should migrate to a different solution such as using a [different section in the docstring](https://numpydoc.readthedocs.io/en/latest/format.html) and/or a tutorial so that the docstring extended summary is not diluted for people viewing it in Jupyter, VSCode, etc.

Reply:
Yep, agree completely. An illustrative geo-data processing tutorial for `nearneighbor` and other functions like `blockmedian` et al would be great. Something for pygmt v0.5.1 perhaps :wink: